### DISCUSSION TITLE
irsdk Global Flags

### DISCUSSION THREAD

**Discussion by badebahr** [kutu/pyirsdk#58] [General]
I am currently developing my own overlay software and checked the last lap and race finished states via the global flags. It worked up to the last season, even though I got really large numbers when outputting the SessionFlag variable, which are not the same as given in the documentation. Since this season it seems the last lap flag does not work anymore. Can you have a look into this?

**Comment by badebahr** [comment]
For a race I recorded the data for I currently get the following 'SessionFlags':
[ 268435456  268435968  268697600  268697601  268697602  268697604
  268697636  268697856  268698112  268698113  269221888  537133568
 1074004480 2147745796 2147746308]
These also seem to change depending on which dataset I have.
I hope this helps. Maybe you are reading out too many bytes?

**Comment by kutu** [comment]
![image](https://github.com/user-attachments/assets/c699d424-3067-4d31-8b8b-1fb66dfb08c4)

**Comment by badebahr** [comment]
Yes these are the starting lights, but how do I get the value for the white flag, checkered etc.? They should be 0x0002 or 0x0001 respectivley, but they do not appear in the list above.

**Comment by kutu** [comment]
https://github.com/kutu/pyirsdk/issues/9

**Comment by theissondergaard** [comment]
@badebahr Did you figure this out? I'm having somewhat of the same challenge

**Reply by hmiefert** [reply]
You can convert the decimal numbers to hexadecimal and then check which of the flags are set.

Check the Flags class in irsdk.py:
268435456 = 0x10000000 = start_hidden
268435968 = 0x10000200 = start_hidden + one_lap_to_green
268697600 = 0x10040000 = start_hidden + serviceable
2147746308 = 0x80040204 = start_go + serviceable + one_lap_to_green + green

At least that's how I understand it - didn't test it myself though.

*edit: it's a long time ago, since I dabbled in pyirsdk, so this might be wrong, but AFAIK the CarIdxSessionFlags is a list and your own car is always on index 0. Maybe try something like this:

```
if len(ir['CarIdxSessionFlags']) >= 1 and ir['CarIdxSessionFlags'][0] & 0x010000:
    print('black flagged')
```

**Reply by theissondergaard** [reply]
Ahhhh, they combine, of course. Printing them out in Hex made me understand. Thanks for kicking my brain into gear :)
